src/Symfony/Component/DependencyInjection/Loader/YamlFileLoader.php

```diff
@@ -432,4 +458,24 @@ private function loadFromExtensions($content)
             $this->container->loadFromExtension($namespace, $values);
         }
     }
+
+    /**
+     * Checks the keywords used to define a service.
+     *
+     * @param string $id         The service name
+     * @param array  $definition The service definition to check
+     * @param string $file       The loaded YAML file
+     */
+    private static function checkDefinition($id, array $definition, $file)
+    {
+        foreach ($definition as $key => $value) {
+            if (!isset(static::$keywords[$key])) {
+                @trigger_error(sprintf('The configuration key "%s" is unsupported for service definition "%s" in "%s". Allowed configuration keys are "%s". The YamlFileLoader object will raise an exception instead in Symfony 4.0 when detecting an unsupported service configuration key.', $key, $id, $file, implode('", "', static::$keywords)), E_USER_DEPRECATED);
+                // @deprecated Uncomment the following statement in Symfony 4.0
+                // and also update the corresponding unit test to make it expect
+                // an InvalidArgumentException exception.
+                //throw new InvalidArgumentException(sprintf('The configuration key "%s" is unsupported for service definition "%s" in "%s". Allowed configuration keys are "%s".', $key, $id, $file, implode('", "', static::$keywords)));
```

Comment:
The drawback would be that we would have to maintain it already, and people would start using it in their composer projects. So I'm -1 on it.

When we start the 4.0 development, we will look at any existing ``@trigger_error`` to convert them to the new code (either removing code, or doing other actions).

I would even remove the commented exception and just put a comment saying it should be converted to an exception in 4.0. This way, the exception message will be written at the 4.0 time, not now, and so using the uptodate deprecation message.